cmd/main.go

```diff
@@ -205,14 +201,32 @@ func addHealthChecks(mgr manager.Manager) {
 	}
 }
 
-func runKymaMetricsCleanup(kymaMetrics *metrics.KymaMetrics, kcpClient client.Client,
-	cleanupIntervalInMinutes int,
-) {
+func cleanupStoredVersions(crdVersionsToDrop string, mgr manager.Manager) {
+	if crdVersionsToDrop == "" {
+		return
+	}
+
+	ctx := context.Background()
+	if !mgr.GetCache().WaitForCacheSync(ctx) {
+		setupLog.V(log.InfoLevel).Error(errFailedToDropStoredVersions, "failed to sync cache")
```

Comment:
I suggest having the `err` itself here since the msg is already failed to sync cache.

And the same for the metrics cleanup error